config/environments/production.rb

```diff
@@ -34,8 +34,8 @@
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
-  # Compress JavaScripts and CSS.
-  config.assets.js_compressor = :uglifier
+  # Compress JavaScripts and CSS (Using ES6 syntax).
+  config.assets.js_compressor = Uglifier.new(harmony: true)
```

Comment:
Aha - ok so here we are missing something as it created this error:

```
Run bundle exec rake assets:precompile
rake aborted!
NameError: uninitialized constant Uglifier
/home/runner/work/spectral-workbench/spectral-workbench/config/environments/production.rb:38:in `block in <top (required)>'
```

https://github.com/publiclab/spectral-workbench/pull/672/checks?check_run_id=3109251932#step:5:7

Do you need to `include Uglifier` or something?

Reply:
This is actually supposed to make the uglifier compatible with the ES6 syntax used in `new-capture.js` according to https://github.com/lautis/uglifier/issues/127 but still the problem remains unsolved...😕

Reply:
So this is a new error `uninitialised constant Uglifier` which should be resolve by https://stackoverflow.com/questions/30338860/heroku-push-error-nameerror-uninitialized-constant-uglifierversion-on-rake but still the problem is persisting.

Reply:
Hm, they note:

> The problem seemed to be an incompatibility between Sprockets 3.1 and the Uglifier compressor version I was using, 2.1. Upgrading Uglifier by putting this line in my Gemfile resolved the issue:

> `gem 'uglifier', '~> 2.7'`

Have you tried that? Else possibly @noi5e may be able to help a bit (or at least have some ideas?) as he's worked a lot with Rails precompilation in his React project!

Reply:
OK here is another possibility:

https://github.com/sstephenson/sprockets/issues/740

> Installing the gem "therubyracer" and rolling uglifier back to 2.6.1 in the Gemfile ( gem 'uglifier', '~> 2.6.1' ) fixes the problem.

...i think we removed therubyracer? Yes - 

https://github.com/publiclab/spectral-workbench/blob/8f89e0fffd0ac02068857ed6fb775aa5a4460591/Gemfile#L16-L18

Try turning it back on and then pegging the uglifier gem to v2.6.1?

Reply:
For comparison, in plots2 -- 

https://github.com/publiclab/plots2/blob/4e52e2ea2875593734b96bac4e8326e058a25d8e/config/environments/production.rb#L18